gui/default/syncthing/core/syncthingController.js

```diff
@@ -646,7 +646,8 @@ angular.module('syncthing.core')
         $scope.probeRemoteGUIAddress = function (deviceId, address) {
             $http({
                 method: "OPTIONS",
-                url: address,
+                // Apply RFC6874 encoding for IPv6 link-local zone identifier
+                url: address.replace('%', '%25'),
```

Comment:
There is probably a proper urlEncode method somewhere.

Reply:
Yes, there is `encodeURI()`, which wrongfully replaces the brackets `[` and `]` for the IPv6 literal as well.  Couldn't use that, so I went with the replacement exactly as the RFC defines it.

Reply:
By the way I also tried stripping off the `%zone-id` part completely.  Under Firefox, that made the JS error messages go away, but still no luck with the actual HTTP request, which is either blocked by policy or cannot work without a zone identifier.  Curl only works when the zone ID is contained, either percent-encoded or not.  What a mess.